packages/ozone-helper/ozone-type/ozone/model/FlowrWarningParsable.ts

```diff
@@ -0,0 +1,6 @@
+import {UUID} from "./Item";
+
+export class FlowrWarningParsable {
+	[key:string]: any
```

Comment:
It means it can have any property, in addition to the one declared in the children types, it don't know if it's what you intended

Reply:
My goal was to have the properties accessible by string.
Example : 
`const propertyName = "id"
const itemId = item[propertyName]`
If you have a better solution, I'm all ears :)

Reply:
Then I do not understand the point of this type (interface). This means that except for the id you can have
```
const foo = getParsable() as FlowrWarningParsable
const bar = foo['anykindofproperty'] // <-- will be "any" but will not throw a Typescript error
```
What's the point of typing if in the end we only work with `any` values ?
To achieve a similar result, we could use a union type and add the new classes to this union type instead of having them implement the completely generic interface`FlowrWarningParsable`.
```
type FlowrWarningParsable = FlowrWarningOperator | LocationBuilding | LocationSite

// Then we can use user defined type guards
function isLocationSite(parsable: FlowrWarningParsable): parsable is LocationSite {
  // ? return parsable.type === 'flowr.location.site' ?
}

// ...and so on
```

( this is just a proposal )

Reply:
What about the index signatures ?
I'm not sure what should be done here, and I think it was the point of the `FlowrWarningParsable` original implementation

Reply:
Declaring an index signature on a type implies that this object can hold any property name but they will all have the same type. So when using an object with different types there is no point in using index signature.
Here, all values must be of the same type. Compiler throws an error if it's not the case:
![Screenshot 2020-10-19 at 10 58 48](https://user-images.githubusercontent.com/5877409/96423875-3befce80-11fa-11eb-879a-7976f89523ba.png)

I do believe that this allows the same result in the end
![Screenshot 2020-10-19 at 11 03 14](https://user-images.githubusercontent.com/5877409/96424229-b7518000-11fa-11eb-8f57-70594761c1aa.png)

Reply:
Thanks for this explaination, it thought it always needed an index signature to be able to index an object with a string
But i guess it was not properly typed when I had these kind of issues